### PR TITLE
Add tests for cleos new option --abi-file 📦

### PR DESCRIPTION
**Pull request by nickjjzhao**
## Change Description
[EPE-1551](https://blockone.atlassian.net/browse/EPE-1551)

Add test cases for `cleos` new option `--abi-file` that was added in [PR#10821](https://github.com/EOSIO/eos/pull/10821) to specify one or more local abi files for cleos for serialzation/deserialization offline, which doesn't require nodeos RPC endpoint.

## Change Type
**Select *ONE*:**
- [ ] Documentation

- [ ] Stability bug fix

- [x] Other

- [ ] Other - special case

## Testing Changes
**Select *ANY* that apply:**
- [ ] New Tests

- [x] Existing Tests

- [ ] Test Framework

- [ ] CI System

- [ ] Other

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

File: tests/cli_test.py [hunk 1]
```diff
@@ -56,6 +57,203 @@ def cli11_optional_option_arg_test():
                                       '-c', chain, '-k', key, '{}'])
     assert(b'signatures' in output)
 
+    """Test option --abi-file """
```
Comment (linhuang-blockone) [inline tests/cli_test.py:137]: Place those new tests in a separate function/test-case.

File: tests/cli_test.py [hunk 2]
```diff
@@ -56,6 +57,203 @@ def cli11_optional_option_arg_test():
...
+    except subprocess.CalledProcessError as ex:
+        print(ex.output)
+    assert(b'please specify --abi-file in form of <contract name>:<abi file path>.' in errs)
+
+    # pack token transfer data
+    account = 'eosio.token'
+    action = 'transfer'
+    unpacked_action_data = '{"from":"aaa","to":"bbb","quantity":"10.0000 SYS","memo":"hello"}'
+    packed_action_data = '0000000000008c31000000000000ce39a08601000000000004535953000000000568656c6c6f'
+    cmd = ['./programs/cleos/cleos', '--abi-file', token_abi_file_arg, 'convert', 'pack_action_data', account, action, unpacked_action_data]
```
Comment (linhuang-blockone) [inline tests/cli_test.py]: The purpose of the new option --abi-file is to provide a capability to do conversion offline. To make sure not use local Nodeos, use a non-existing port in "-u" option of cleos like "cleos -u 127.0.0:12345".

**Review by linhuang-blockone — CHANGES_REQUESTED** [review]
Use "cleos -u 127.0.0.1:12345" to make sure not use local nodeos.

File: tests/cli_test.py (hunk 1 above)
Comment (nickjjzhao) [reply tests/cli_test.py:137]: Fixed.

File: tests/cli_test.py (hunk 2 above)
Comment (nickjjzhao) [reply tests/cli_test.py]: Fixed.

**Review by linhuang-blockone — APPROVED** [review]

**Review by linhuang-blockone — APPROVED** [review]

**Pull request merged**